### ISSUE TITLE
[LSP] completion use wrong column offset

### ISSUE THREAD

**Issue by SolaWing** [sorbet/sorbet#2489]
#### Input

support following code
```ruby
# typed: true
# frozen_string_literal: true

Array.ne # cursor is between n|e, offset is 7
```

with LSP request:
```json
{
	"id": 6,
	"jsonrpc": "2.0",
	"method": "textDocument/completion",
	"params": {
		"position": {
			"character": 7,
			"line": 3
		},
		"textDocument": {
			"uri": "xxxx.rb"
		}
	}
}
```

#### Observed output

lsp server return:

```json
{
	"jsonrpc": "2.0",
	"id": 6,
	"requestMethod": "textDocument/completion",
	"result": {
		"isIncomplete": false,
		"items": [
			{
				"label": "new",
				"kind": 2,
				"detail": "Class#new",
				"documentation": {
					"kind": "markdown",
					"value": "```ruby\\nsig {params(args: T.untyped).returns(T.untyped)}\\ndef new(*args); end\\n```\\n\\n---\\n\\nCalls `allocate` to create a new object of *class*'s class, then invokes\\nthat object's `initialize` method, passing it *args*. This is the method\\nthat ends up getting called whenever an object is constructed using .new."
				},
				"sortText": "000001",
				"insertTextFormat": 1,
				"textEdit": {
					"range": {
						"start": {
							"line": 3,
							"character": 5
						},
						"end": {
							"line": 3,
							"character": 7
						}
					},
					"newText": "new"
				}
			}
		]
	}
}

```

#### Expected behavior

textEdit range is wrong, should be 6-7 or 6-8.

#### Other issues:

`Array.` have no completions, must provide query to get completion. expect to return all completions without filter

**Comment by jez** [comment]
> `Array.` have no completions, must provide query to get completion. expect to return all completions without filter

This is a known issue. It's unlikely to get fixed unless someone more talented with parser generators than me looks at how to make our parser recover from parse errors better. That would be a welcome change.

**Comment by jez** [comment]
This was fixed in #5081 

Thanks again for the report.
